.github/workflows/continuous-integration.yml

```diff
@@ -28,4 +28,7 @@ jobs:
         run: pytest tests -v
 
       - name: Lint
-        run: pre-commit run --all-files
\ No newline at end of file
+        run: pre-commit run --all-files
+
+      - name: Type check
+        run: mypy .
```

Comment:
I'm still quite confused by hatch. This will use the hatch env? Even without running something like `hatch run mypy .`?

Reply:
Mmm I don't use any `hatch` commands directly locally. My shell automatically activates the virtulaenv when I `cd` into the directory.

In the case of the GHA, I don't think there's a virtulenv at all. Just running stuff in some python interpreter.

I think our only hard requirement on using hatch is at build time (and even that is hidden by `python -m build .`, which figures out how to use hatch for building the wheel and sdit based on the metadata in pyproject.toml).

But in general: we need a contributing guide.

Reply:
I see. I'm a big fan of Poetry and Pixi but have never used hatch before, so I'm learning it.

Reply:
I like it, in that there's (ideally) nothing new to learn if you don't want to :)